adlfs/spec.py

```diff
@@ -135,6 +135,11 @@ class AzureBlobFileSystem(AsyncFileSystem):
         The storage account key. This is used for shared key authentication.
         If any of account key, sas token or client_id is specified, anonymous access
         will be used.
+    account_host: str
```

Comment:
We'll have to be careful with the naming here.

Seems that Azure already uses "account host" to mean something slightly different. From https://learn.microsoft.com/en-us/azure/storage/blobs/storage-custom-domain-name?tabs=azure-portal#step-1-get-the-host-name-of-your-storage-endpoint

> The host name is the storage endpoint URL without the protocol identifier and the trailing slash.

I'd much prefer it if adlfs just took the `account_url` like the underlying `azure.storage.blob` library.

Then maybe in `__init__` we use

```
self._parsed = urllib.parse.urlparse(account_url)
```

which is something like

```python
In [4]: urllib.parse.urlparse("https://foo.blob.core.windows.net")
Out[4]: ParseResult(scheme='https', netloc='foo.blob.core.windows.net', path='', params='', query='', fragment='')
```

And then we store the pieces we need later on for `connect`, which I think is

- `scheme`
- `netloc.split(".", 1)[-1]`, e.g. `blob.core.windows.net`

Thoughts on that?

Reply:
The account_host should be exactly the format you mentioned, as I attempted to describe in the parameter description. You can see that "https:/"/ is added in L481: "self.account_url: str = f"https://{self.account_host}".

I include the account_host parameter to mimic what's already in the code.

Reply:
Oh sorry, read the docs you wrote complete backwards :)